test/e2e/k8s_util.go

```diff
@@ -159,17 +160,23 @@ func (k *KubernetesUtils) probe(
 	// It needs to check both err and stderr because:
 	// 1. The probe tried 3 times. If it checks err only, failure+failure+success would be considered connected.
 	// 2. There might be an issue in Pod exec API that it sometimes doesn't return error when the probe fails. See #2394.
+	var actualResult PodConnectivityMark
 	if err != nil || stderr != "" {
 		// log this error as trace since may be an expected failure
 		log.Tracef("%s -> %s: error when running command: err - %v /// stdout - %s /// stderr - %s", podName, dstName, err, stdout, stderr)
 		// If err != nil and stderr == "", then it means this probe failed because of
 		// the command instead of connectivity. For example, container name doesn't exist.
 		if stderr == "" {
-			return Error
+			actualResult = Error
 		}
-		return DecideProbeResult(stderr, 3)
+		actualResult = DecideProbeResult(stderr, 3)
+	} else {
+		actualResult = Connected
+	}
+	if expectedResult != nil && *expectedResult != actualResult {
+		log.Infof("%s -> %s: expected %s but got %s: err - %v /// stdout - %s /// stderr - %s", podName, dstName, *expectedResult, actualResult, err, stdout, stderr)
```

Comment:
`log.Tracef` might be more suitable than `log.Infof` here, also keeping consistency in this file.

Reply:
I think that means we can only see the log when the test itself changes the log level to Trace, which could cause flood of logs generated by the matrix?
The point of the new log is that it only logs the error when it doesn't match the expectation. So if everything works fine, there is no verbose trace logs. If one probe fails, we can find detailed outputs with this probe only.

Reply:
Got it, makes sense!